src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceClientTestClassComposer.java

```diff
@@ -97,41 +89,22 @@
 import org.junit.BeforeClass;
 import org.junit.Test;
 
-public class ServiceClientTestClassComposer implements ClassComposer {
-  private static final Statement EMPTY_LINE_STATEMENT = EmptyLineStatement.create();
+public abstract class AbstractServiceClientTestClassComposer implements ClassComposer {
+  protected static final Statement EMPTY_LINE_STATEMENT = EmptyLineStatement.create();
 
-  private static final String CHANNEL_PROVIDER_VAR_NAME = "channelProvider";
-  private static final String CLASS_NAME_PATTERN = "%sClientTest";
-  private static final String CLIENT_VAR_NAME = "client";
-  private static final String GRPC_TESTING_PACKAGE = "com.google.api.gax.grpc.testing";
+  protected static final String CLIENT_VAR_NAME = "client";
   private static final String MOCK_SERVICE_VAR_NAME_PATTERN = "mock%s";
   private static final String PAGED_RESPONSE_TYPE_NAME_PATTERN = "%sPagedResponse";
-  private static final String SERVICE_HELPER_VAR_NAME = "mockServiceHelper";
-  private static final String STUB_SETTINGS_PATTERN = "%sSettings";
-
-  private static final ServiceClientTestClassComposer INSTANCE =
-      new ServiceClientTestClassComposer();
-
-  private static final TypeStore FIXED_TYPESTORE = createStaticTypes();
-  private static final TypeNode LIST_TYPE =
-      TypeNode.withReference(ConcreteReference.withClazz(List.class));
-  private static final TypeNode MAP_TYPE =
-      TypeNode.withReference(ConcreteReference.withClazz(Map.class));
-  private static final TypeNode RESOURCE_NAME_TYPE =
-      TypeNode.withReference(
-          ConcreteReference.withClazz(com.google.api.resourcenames.ResourceName.class));
-
-  private static final AnnotationNode TEST_ANNOTATION =
-      AnnotationNode.withType(FIXED_TYPESTORE.get("Test"));
-  // Avoid conflicting types with com.google.rpc.Status.
-  private static final TypeNode GRPC_STATUS_TYPE =
-      TypeNode.withReference(
-          ConcreteReference.builder().setClazz(io.grpc.Status.class).setUseFullName(true).build());
-
-  private ServiceClientTestClassComposer() {}
-
-  public static ServiceClientTestClassComposer instance() {
-    return INSTANCE;
+
+  private final TypeStore fixedTypeStore;
+  private final ClassNames classNames;
+  private final AnnotationNode testAnnotation;
+
+  protected AbstractServiceClientTestClassComposer(
+      TypeStore fixedTypeStore, ClassNames classNames) {
+    this.fixedTypeStore = fixedTypeStore;
+    this.classNames = classNames;
+    this.testAnnotation = AnnotationNode.withType(fixedTypeStore.get("Test"));
```

Comment:
I can move it back to the `public static final` but that would mean recreation of `TypeNode` in a more custom way. I was under impression that there was a standard approach to declaring the set of  `TypeNode` elements in each composer - the `creteTypes()` or `createStaticTypes()` (different composers have different name). I don't see big differfence between the `testAnnotaion` here and any other type created in the `createType()` methods by the composers. 

I'm wondering why not just remove `testAnnotation` field completely and access it as `getFixedTypeStore("Test")` instead everywhere where it is needed? Note, this approach (`FIXED_TYPESTORE.get(name)` or `getFixedTypeStore().get(name)`) has already been used heavily in the composers implementation.

There is no guarantee that the `"Test"` class will exist in `fixedTypeStore`, but the same is true for everything else in `fixedTypeStore` (so again, it does not feel like there is anything special about `"Test"` class compared to all the others in `fixedTypeStore`).

Note, in terms of when the failure can be detected, `private static final FIXED_TYPESTORE = createTypes()` and `private final fixedTypeStore`, initialized in the composer's constructor - they are identical, because `fixedTypeStore` is initialized in the constructor of the singleton classes (all composers are singletons), the `INSTANCE` is `private static final` variable in the classes, and as a result is instantiated in the same phase as the `private static final FIXED_TYPESTORE`. I.e. it is all the same thing as before, just one extra frame in the call stack (`fixedTypeStore` will be instantiated in the composer's class constructor stack, but still on the class initialization phase, basically right after loading the class in the memory.